File: PPLT2/Exceptions.py
# ########################################################################### #
# This is part of the PPLT project. PPLT is a framework for master-slave      #
# based communication.                                                        #
# Copyright (C) 2003-2006 Hannes Matuschek <hmatuschek@gmx.net>               #
#                                                                             #
# This library is free software; you can redistribute it and/or               #
# modify it under the terms of the GNU Lesser General Public                  #
# License as published by the Free Software Foundation; either                #
# version 2.1 of the License, or (at your option) any later version.          #
#                                                                             #
# This library is distributed in the hope that it will be useful,             #
# but WITHOUT ANY WARRANTY; without even the implied warranty of              #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU            #
# Lesser General Public License for more details.                             #
#                                                                             #
# You should have received a copy of the GNU Lesser General Public            #
# License along with this library; if not, write to the Free Software         #
# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA   #
# ########################################################################### #


class Error (Exception):
    """ This is the general base-exception for the PPLT2 project.
 all PPLT-code and modules will raised (should raise) this or a
 exception derived from this. """



class InternalError (Error):
    """ This is the base-exception for all internal errors.
 If this exception or one derived from this will indicated an serious
 internal error. Mostly an indicatior for mad code!"""



class ModuleError (Error):
    """ This is the base-exception for all module related errors."""



class ItemNotFound (Error):
    """ This exception will be raised if an item (symbol, module, ...) can't 
 be found."""



class ItemBusy (Error):
    """ This exception will be raised if an item (module) is busy. Also it 
    will be used for a time-out-error."""

File: PPLT2/Exceptions.cs
using System;

namespace PPLTExceptions{

/// <summary>This is the base exception for all PPLT specific 
/// Errors.</summary>
class ErrException: Exception{ 
    public ErrException():base(){}
    public ErrException(string Message):base(Message){}
    public ErrException(string Message, params object[] mods):
        base(String.Format(Message,mods)){}
}

/// <summary>This exception will be raised if a "item" ie a symbol, 
/// module... can't be found.</summary>
class ErrItemNotFound: ErrException{
    public ErrItemNotFound():base(){}
    public ErrItemNotFound(string Message):base(Message){}
    public ErrItemNotFound(string Message, params object[] mods):
        base(Message, mods){}
}

/// <summary>This exception will be raise if an item
/// like a module is used for a longer time than the TTL
/// of the message</summary>    
class ErrItemBusy: ErrException { 
    public ErrItemBusy():base(){}
    public ErrItemBusy(string Message):base(Message){}
    public ErrItemBusy(string Message, params object[] mods):
        base(Message, mods){}
}

/// <summary>This exception will be raised if a error accours in a module
/// </summary>  
class ErrModule: ErrException {
    public ErrModule():base(){}
    public ErrModule(string Message):base(Message){}
    public ErrModule(string Message, params object[] mods):
        base(Message,mods){}
}

/// <summary>This exception will be raised if a message can't be processed
/// before his TTL.</summary>
class ErrTimeOut: ErrException {
    public ErrTimeOut():base(){}
    public ErrTimeOut(string Message):base(Message){}
    public ErrTimeOut(string Message, params object[] mods):
        base(Message, mods){}
}


} // END OF NAMESPACE PPLTEXCEPTIONS
